go/tasks/pluginmachinery/utils/dns.go

```diff
@@ -0,0 +1,25 @@
+package utils
+
+import (
+	"regexp"
+	"strings"
+)
+
+var dns1123MaxLen = 63
+var dns1123InvalidRegex = regexp.MustCompile("[^-a-z0-9]")
+var camelCaseRegex = regexp.MustCompile("([a-z0-9])([A-Z])")
+
+func ConvertToDNS1123CompatibleString(name string) string {
+	name = ConvertCamelCaseToKebabCase(name) // best effort to preserve readability for Java class name
+	name = strings.ToLower(name)
+	name = dns1123InvalidRegex.ReplaceAllString(name, "")
+	name = strings.Trim(name, ".-")
+	if len(name) > dns1123MaxLen {
+		name = name[:dns1123MaxLen]
```

Comment:
For uniqueness purposes, do you want to cut it even shorter and hash the rest?
Check here for examples: https://github.com/flyteorg/flytepropeller/blob/master/pkg/utils/encoder.go

Reply:
Good idea. `encoder.go` is in flytepropeller, so I suppose I can move everything to flytepropeller. I did it here in flyteplugins because validation also happens in plugin. So there's a chance that the pod name conversion helper is also needed in flyteplugins.

Reply:
Code is moved to here https://github.com/flyteorg/flytepropeller/pull/325

Reply:
Apologies I didn't mean to suggest moving the code there... just pointing you at examples that sort of overlap with what you are trying to do...

I'm ok actually keeping that code here.. (I would go as far as saying I prefer the code be here...)

Reply:
They're moved back (among other changes)